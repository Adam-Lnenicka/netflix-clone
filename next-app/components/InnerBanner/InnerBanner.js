import React, { useState } from "react";
import { useDispatch } from "react-redux";
import Link from "next/link";

import { searchMovieTitleCreator } from "../../store/actionCreators";

const InnerBanner = () => {
  const dispatch = useDispatch();
  const [searchTerm, setSearchTerm] = useState("");
  return (
    <div className="banner">
      <div className="inner-banner">
        <Link href="/add-movie">
          <a>
            <div className="button-main add-button">Add Movie</div>
          </a>
        </Link>

        <h1>FIND YOUR MOVIE</h1>
        <div className="inner-banner__form">
          <input
            type="text"
            placeholder="What do you want to watch"
            onChange={(e) => {
              setSearchTerm(e.target.value);
              console.log(searchTerm);
            }}
            // onChange={(e) => dispatch(searchMovieTitleCreator(e.target.value))}
          />
          <button
            className="button-main"
            onClick={() => {
              dispatch(searchMovieTitleCreator(searchTerm));
            }}
          >
            search
          </button>
        </div>
      </div>
    </div>
  );
};
export default InnerBanner;
