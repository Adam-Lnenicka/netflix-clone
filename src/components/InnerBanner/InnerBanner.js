import React, { useState } from "react";

// import NewMovie from "../NewMovie/NewMovie";

const InnerBanner = () => {
  // const dispatch = useDispatch();
  const [searchTerm, setSearchTerm] = useState("");
  return (
    <div className="banner">
      <div className="inner-banner">
        {/* <Link to="/add-movie">
          <div className="button-main add-button">Add Movie</div>
        </Link> */}

        <h1>FIND YOUR MOVIE</h1>
        <div className="inner-banner__form">
          <input
            type="text"
            className="input-box"
            placeholder="What do you want to watch"
            onChange={(e) => {
              setSearchTerm(e.target.value);
              console.log(searchTerm);
            }}
            // onChange={(e) => dispatch(searchMovieTitleCreator(e.target.value))}
          />
          <button
            className="button-main"
            data-test="submit-button"
            onClick={() => {
              // dispatch(searchMovieTitleCreator(searchTerm));
            }}
          >
            search
          </button>
        </div>
      </div>
    </div>
  );
};
export default InnerBanner;
