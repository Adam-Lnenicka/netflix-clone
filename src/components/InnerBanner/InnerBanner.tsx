import React, { useState } from "react";
import { useDispatch } from "react-redux";
import { searchMovieTitleCreator } from "../../store/actionCreators";
import { Link } from "react-router-dom";
// import NewMovie from "../NewMovie/NewMovie";

const InnerBanner: React.FC = () => {
  const dispatch = useDispatch();
  const [searchTerm, setSearchTerm] = useState("");
  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setSearchTerm(e.target.value);
    console.log(searchTerm);
  };
  return (
    <div className="banner">
      <div className="inner-banner">
        <Link to="/add-movie">
          <div className="button-main add-button">Add Movie</div>
        </Link>

        <h1>FIND YOUR MOVIE</h1>
        <div className="inner-banner__form">
          <input
            type="text"
            placeholder="What do you want to watch"
            onChange={handleChange}
            // onChange={(e) => dispatch(searchMovieTitleCreator(e.target.value))}
          />
          <button
            className="button-main"
            onClick={() => {
              dispatch(searchMovieTitleCreator(searchTerm));
            }}
          >
            search
          </button>
        </div>
      </div>
    </div>
  );
};
export default InnerBanner;
