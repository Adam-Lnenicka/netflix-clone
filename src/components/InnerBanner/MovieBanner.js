import React from "react";
import { useParams } from "react-router";
import { useApiMoviesArraySelector } from "../../store/selectors";
import { ButtonExit } from "../../styles/global/Buttons";
import Shorten from "../Shorten";
import {
  MovieContainer,
  MovieDetailsImage,
  MovieRating,
  Runtime,
} from "./styles/MovieBannerStyles";

const MovieBanner = () => {
  const { movieId } = useParams();
  const apiMoviesArray = useApiMoviesArraySelector();
  const thisMovie = apiMoviesArray.map((movie) =>
    movie.find((movie) => movie.id == movieId)
  );

  return (
    <MovieContainer>
      <MovieDetailsImage
        src={thisMovie.map((m) => m.poster_path)}
        alt={thisMovie.map((m) => m.title)}
        onError={(e) =>
          (e.target.src =
            "https://images.unsplash.com/photo-1560109947-543149eceb16?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1035&q=80")
        }
      />
      <div>
        <div className="flex">
          <ButtonExit to="/">X</ButtonExit>
          <h2> {thisMovie.map((m) => m.title)}</h2>
          <MovieRating> {thisMovie.map((m) => m.vote_average)}</MovieRating>
        </div>
        <p>{thisMovie.map((m) => m.tagline)}</p>
        <Runtime>
          <p> {Shorten(thisMovie.map((m) => m.release_date))} </p>
          <p> {thisMovie.map((m) => m.runtime)} min</p>
        </Runtime>
        <p>{thisMovie.map((m) => m.overview)}</p>
      </div>
    </MovieContainer>
  );
};

export default MovieBanner;
