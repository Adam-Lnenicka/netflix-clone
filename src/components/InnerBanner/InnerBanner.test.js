import "jsdom-global/register";

import React from "react";
import toJSON from "enzyme-to-json";
import { mount, shallow } from "enzyme";
import InnerBanner from "./InnerBanner";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

describe("Inner Banner", () => {
  const initialState = {};
  const mockStore = configureStore();
  const mockFunction = jest.fn();

  it("Simulates dispatch on search function", () => {
    const store = mockStore(initialState);

    const wrapper = mount(
      // <Provider store={store}>
      <InnerBanner onCLick={mockFunction} />
      // </Provider>
    );
    wrapper.find("button").simulate("click");
    expect(mockFunction).toBeCalle();
  });

  it("submit button displays", () => {
    // const store = mockStore(initialState);

    const wrapper = mount(
      // <Provider store={store}>
      <InnerBanner />
      // </Provider>
    );

    const findByTestAttr = (wrapper, val) => {
      return wrapper.find(`[data-test="${val}"]`);
    };
    // test('submit button does not display', () => {
    const submitButton = findByTestAttr(wrapper, "submit-button");
    expect(submitButton.exists()).toBe(true);
    // });
  });

  it("state updates with value of input box upon change", () => {
    const wrapper = mount(
      // <Provider store={store}>
      <InnerBanner />
      // </Provider>
    );
    const findByTestAttr = (wrapper, val) => {
      return wrapper.find(`[data-test="${val}"]`);
    };
    const inputBox = findByTestAttr(wrapper, "input-box");
    const mockEvent = { target: { value: "input" } };

    inputBox.simulate("change", mockEvent);
    // expect(mockSetCurrentGuess).toHaveBeenCalledWith("input");
  });

  it("matches snapshot", () => {
    // const store = mockStore(initialState);

    const wrapper = shallow(
      // <Provider store={store}>u
      <InnerBanner />
      // </Provider>
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
