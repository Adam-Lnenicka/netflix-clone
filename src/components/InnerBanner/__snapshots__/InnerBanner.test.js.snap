// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inner Bannner shallow wrapper Inner Banner matches snapshot 1`] = `
<div
  className="banner"
>
  <div
    className="inner-banner"
  >
    <h1>
      FIND YOUR MOVIE
    </h1>
    <div
      className="inner-banner__form"
    >
      <input
        className="input-box"
        onChange={[Function]}
        placeholder="What do you want to watch"
        type="text"
      />
      <button
        className="button-main"
        data-test="submit-button"
        onClick={[Function]}
      >
        search
      </button>
    </div>
  </div>
</div>
`;
